diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 9fedd83307439..4ef0925545e31 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1868,6 +1868,8 @@ static_library("browser") {
   ]
 
   public_deps = [
+    # 加入我们的依赖
+    "//demo/demo_mojo_v8:browser",
     "//base",
     "//chrome/common",
     "//chrome/common:buildflags",
diff --git a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
index 5feac0c791c70..7cbd4bb5d9e31 100644
--- a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
+++ b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
@@ -73,6 +73,8 @@
 #endif
 #endif
 
+#include "demo/demo_mojo_v8/browser/demo_impl.h"
+
 namespace {
 
 // Helper method for ExposeInterfacesToRenderer() that checks the latest
@@ -134,6 +136,10 @@ void ChromeContentBrowserClient::ExposeInterfacesToRenderer(
     service_manager::BinderRegistry* registry,
     blink::AssociatedInterfaceRegistry* associated_registry,
     content::RenderProcessHost* render_process_host) {
+
+  // 注册 DemoImpl
+  demo::DemoImpl::Initialize(registry);
+
   // The CacheStatsRecorder is an associated binding, instead of a
   // non-associated one, because the sender (in the renderer process) posts the
   // message after a time delay, in order to rate limit. The association
diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index cd4395923de75..a739f6e632fa3 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -145,6 +145,8 @@ static_library("renderer") {
   public_deps = [ "//components/contextual_search:buildflags" ]
 
   deps = [
+    #加入我们的依赖
+    "//demo/demo_mojo_v8:renderer",
     "//base/allocator:buildflags",
     "//build:chromeos_buildflags",
     "//chrome:resources",
diff --git a/chrome/renderer/chrome_content_renderer_client.cc b/chrome/renderer/chrome_content_renderer_client.cc
index ceb29f83a9dc1..c7fe313e195fc 100644
--- a/chrome/renderer/chrome_content_renderer_client.cc
+++ b/chrome/renderer/chrome_content_renderer_client.cc
@@ -234,6 +234,8 @@
 #include "chrome/renderer/supervised_user/supervised_user_error_page_controller_delegate_impl.h"
 #endif
 
+#include "demo/demo_mojo_v8/renderer/demo_render_frame_observer.h"
+
 using autofill::AutofillAgent;
 using autofill::AutofillAssistantAgent;
 using autofill::PasswordAutofillAgent;
@@ -467,6 +469,11 @@ void ChromeContentRendererClient::ExposeInterfacesToBrowser(
 
 void ChromeContentRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
+
+    // 创建 Observer 并监听
+    // 这不会导致内存泄漏，因为 render_frame 会通知它释放自己的内存
+    new demo::DemoRenderFrameObserver(render_frame);
+
   ChromeRenderFrameObserver* render_frame_observer =
       new ChromeRenderFrameObserver(render_frame, web_cache_impl_.get());
   service_manager::BinderRegistry* registry = render_frame_observer->registry();
