diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 51efc0986c4b5..dc7f34b5f3a27 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1862,6 +1862,7 @@ static_library("browser") {
     "//chrome/browser/headless",
     "//chrome/browser/profiles",
     "//chrome/browser/ui/tabs:tab_enums",
+    "//demo/demo_mojo_v8:browser",
   ]
 
   deps = [
diff --git a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
index 74d28c413c7e3..43b2a680b4bbe 100644
--- a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
+++ b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
@@ -131,6 +131,8 @@
 #include "chrome/browser/offline_pages/offline_page_tab_helper.h"
 #endif
 
+#include "demo/demo_mojo_v8/browser/demo_impl.h"
+
 namespace {
 
 #if BUILDFLAG(SAFE_BROWSING_AVAILABLE)
@@ -220,6 +222,8 @@ void ChromeContentBrowserClient::ExposeInterfacesToRenderer(
   registry->AddInterface<metrics::mojom::CallStackProfileCollector>(
       base::BindRepeating(&metrics::CallStackProfileCollector::Create));
 
+  demo::DemoImpl::Initialize(registry);
+
   if (NetBenchmarking::CheckBenchmarkingEnabled()) {
     Profile* profile =
         Profile::FromBrowserContext(render_process_host->GetBrowserContext());
diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 283e1c70773a3..952f9813a6dd8 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -235,6 +235,7 @@ static_library("renderer") {
     "//ui/gfx/geometry",
     "//ui/surface",
     "//v8",
+    "//demo/demo_mojo_v8:renderer",
   ]
 
   data_deps = [ "//tools/v8_context_snapshot" ]
diff --git a/chrome/renderer/chrome_content_renderer_client.cc b/chrome/renderer/chrome_content_renderer_client.cc
index a69564cd5dfdf..8bb3ca236d64b 100644
--- a/chrome/renderer/chrome_content_renderer_client.cc
+++ b/chrome/renderer/chrome_content_renderer_client.cc
@@ -256,6 +256,8 @@
 #include "chrome/renderer/media/chrome_key_systems.h"
 #endif
 
+#include "demo/demo_mojo_v8/renderer/demo_render_frame_observer.h"
+
 using autofill::AutofillAgent;
 using autofill::PasswordAutofillAgent;
 using autofill::PasswordGenerationAgent;
@@ -580,6 +582,9 @@ void ChromeContentRendererClient::ExposeInterfacesToBrowser(
 
 void ChromeContentRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
+  // 这不会导致内存泄漏，因为 render_frame 会通知它释放自己的内存
+  new demo::DemoRenderFrameObserver(render_frame);
+
   ChromeRenderFrameObserver* render_frame_observer =
       new ChromeRenderFrameObserver(render_frame, web_cache_impl_.get());
   service_manager::BinderRegistry* registry = render_frame_observer->registry();
